src/Microsoft.Tye.Hosting/DockerImagePuller.cs

```diff
@@ -66,16 +67,39 @@ private async Task PullContainerAsync(string image)
         {
             await Task.Yield();
 
+            string name = image;
+            string version = "latest";
+            var idx = image.LastIndexOf(":");
+            if (image.IndexOf("/") < idx)
+            {
+                name = image.Substring(0, idx);
+                version = image.Substring(idx + 1);
+            }
+
+            string output = string.Empty;
+            await ProcessUtil.RunAsync(
+                             "docker",
+                             "images",
+                             outputDataReceived: data => output += data + '\n',
+                             throwOnError: false);
+
+            var matchCollection = Regex.Matches(output, @"([^\s]+)\s+([^\s]+).+");
```

Comment:
Instead of this, use the --format and --filter parameters.

Reply:
Fixed